lib/server.go

```diff
@@ -1930,12 +1930,19 @@ func (srv *Server) _addNewTxn(
 		return nil, err
 	}
 
-	if srv.blockchain.chainState() != SyncStateFullyCurrent {
+	srv.blockchain.ChainLock.RLock()
+	tipHeight := uint64(srv.blockchain.BlockTip().Height)
```

Comment:
Cache the chain state so it doesn't shift from underneath you
```suggestion
	tipHeight := uint64(srv.blockchain.BlockTip().Height)
	chainState := srv.blockchain.chainState()
```